helper-cli/src/main/kotlin/utils/PathExcludeGenerator.kt

```diff
@@ -51,14 +52,12 @@ internal object PathExcludeGenerator {
      * containing all given [filePaths] which must be relative to the root directory of the source tree.
      */
     fun generateDirectoryExcludes(filePaths: Collection<String>): Set<PathExclude> {
-        val files = filePaths.mapTo(mutableSetOf()) { File(it) }
-        val dirs = getAllDirectories(files)
-
-        val dirsToExclude = mutableMapOf<File, PathExcludeReason>()
+        val dirs = getAllDirectories(filePaths)
+        val dirsToExclude = mutableMapOf<String, PathExcludeReason>()
 
         dirs.forEach { dir ->
             val (_, reason) = PATH_EXCLUDES_REASON_FOR_DIR_NAME.find { (pattern, _) ->
-                FileMatcher.match(pattern, dir.name, ignoreCase = true)
+                FileMatcher.match(pattern, dir.substringAfterLast('/'), ignoreCase = true)
```

Comment:
I find it a bit more convenient to have the helper functions work with `Files` and also more consistent.
Why are you changing these to use `String`? e.g. turn `dir.name` into `dir.substringAfterLast('/')` ?

Would you be ok dropping this commit from this PR and maybe do that a bit later?
I have more changes in my stash, but I'm not yet sure how code will look in the end,
e.g. if there will be further calls to File.getAncestorFiles(), but I think there is a good chance.

Reply:
> I find it a bit more convenient to have the helper functions work with `Files` and also more consistent.

I basically agree, but actually your calling code is operating on strings and your mapping to files, so using the existing function that works with strings actually makes the code simpler.

> Why are you changing these to use `String`?

Note that I'm not changing the existing helper function to take a string, but my goal was to reuse the existing string-based function.

> e.g. turn `dir.name` into `dir.substringAfterLast('/')` ?

I agree this bit looks uglier than before, but IMO it's worth it for the sake of the other simplifications / code deduplications.

> if there will be further calls to File.getAncestorFiles(), but I think there is a good chance.

If your upcoming code would indeed look much uglier with the string-version I'd instead consider to rewrite the common `getAllAncestorDirectories()` utility function to operate on files. There is just one single other called of it in the `RootLicenseMatcher`. And we should really try to not duplicate so much code and work.

Reply:
Could we then please split out this single (in my view controversial commit) into a separate PR, so that I continue my work while we sort this one out?

Reply:
I've dropped the last commit.

PS: These refactoring would not be necessary if you did a more thorough research of what functions we already have in the first place 😉